### DISCUSSION TITLE
What are my limits if I host a page on Github Pages? (I apologize for this overwrought question, but the terms used in some answers are Greek to me)

### DISCUSSION THREAD

**Discussion by neilpat05** [community/community#27714] [New to GitHub] [answered]
At the outset, I am a layperson, not a tech guy, am a doctor by profession. I learned basic web design: HTML, CSS, and some JS for my work and made a website for teaching students pro-bono. I learned that we can host it on GitHub and it's free. Most hosting sites charge for hosting, and I am unable to comprehend how Github does this for free. (Do note that I am a layperson and do not understand a lot of the ToS's mentioned). Could someone dumb it down for me whether I could host my site here on Github for free and what would be my limitations?
- It only contains HTML, CSS, and JS. 
- Probably would attract like 500-1000 visitors for the first 1-2 months initially and later around 200 per month.

Also, if I purchased a domain name from GoDaddy for example, can I host it here?

**Comment by EliasAfara** [comment, marked as answer]
GitHub Pages is a static site hosting service that takes HTML, CSS, and JavaScript files straight from a repository on GitHub, optionally runs the files through a build process, and publishes a website. You can see examples of GitHub Pages sites in the [GitHub Pages examples collection](https://github.com/collections/github-pages-examples).

For your second point, there should not be any problem. You can use CloudFlare or any other CDN in front of the GitHub Page so that whenever there is a spike, it will hit the CDN, not your page directly.

> if I purchased a domain name from GoDaddy for example, can I host it here?

Yes, you can host your site on GitHub's github.io domain or your own custom domain. For more information, see [Using a custom domain with GitHub Pages](https://docs.github.com/en/articles/using-a-custom-domain-with-github-pages). and check [Manage a custom domain](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site).

[Source](https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages)

**Reply by neilpat05** [reply]
Is there any tutorial for using Cloudflare in front of the Github page? Like a walkthrough tutorial for newbies.

**Reply by EliasAfara** [reply]
You can checkout [this](https://github.com/orgs/community/discussions/27118#discussioncomment-3254691) article. 

If you are interested in Netlify instead of Github pages, check [this](https://nooshu.com/blog/2021/09/06/migrating-from-github-pages-to-cloudflare-and-netlify/).

**Reply by ng-druid** [reply]
In my experimentation with static hosting across multiple clouds I found cloudflare to be one of the easiest to wire up. Unfortunately that seems to come with significant expense since cloudflare is also one of the most expensive static site hosts.

The nice thing is that once you develop a mind set of GitHub repos as static site repos most cloud vendors integrate with GitHub fairly easily to host static sites. This makes it very easy to experiment with different vendors or take advantage of individual incentives spreading out sites across multiple clouds.

**Reply by neilpat05** [reply]
Doesn't Cloudflare offer a free plan as well to host a website as well? At least those are not business sensitive.

**Comment by amir1366t** [comment]
Hello. I use my account only to learn and donate to popular projects and I still want to continue. Please reactivate my account so that I can continue to learn.
